services/data-warehouse/setup.sh

```diff
@@ -218,6 +218,36 @@ is_port_in_use() {
     fi
 }
 
+# Open URL in default browser (cross-platform)
+open_browser_url() {
```

Comment:
This worries me.  Have you tested across browsers?  I can test against windows and linux with more effort (i.e. having to launch a VM).